src/core/cubic.c

```diff
@@ -584,6 +593,36 @@ CubicCongestionControlOnDataLost(
     QuicConnLogCubic(QuicCongestionControlGetConnection(Cc));
 }
 
+_IRQL_requires_max_(DISPATCH_LEVEL)
+void
+CubicCongestionControlOnEcn(
+    _In_ QUIC_CONGESTION_CONTROL* Cc,
+    _In_ const QUIC_ECN_EVENT* EcnEvent
+    )
+{
+    QUIC_CONGESTION_CONTROL_CUBIC* Cubic = &Cc->Cubic;
+
+    BOOLEAN PreviousCanSendState = CubicCongestionControlCanSend(Cc);
+
+    //
+    // If the ECN signal is received after the most recent congestion event
+    // (or if there hasn't been a congestion event yet) then treat it as a
+    // new congestion event.
+    //
+    if (!Cubic->HasHadCongestionEvent ||
+        EcnEvent->LargestPacketNumberAcked > Cubic->RecoverySentPacketNumber) {
```

Comment:
CubicCongestionControlOnDataLost compares with LargestPacketNumberLost but here we compare with LargestPacketNumberAcked- is this from the spec?

Reply:
No, the spec only says one congestion event per RTT. For ECN, "Lost" doesn't apply. So, I just use LargestPacketNumberAcked to do the same thing.

Reply:
Oh I get it.